w5e4.py

```diff
@@ -0,0 +1,102 @@
+# "w5e4"
+
+# connected vessels
+'''
+ Write a function called interleave that gets one or more iterable parameters, and returns a list of intertwined members.
+'''
+
+# function to interleave lists
+def interleave(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    #Run like this on the loop because it is important that the first members of each list appear first
+    while j < longest:
+        for i in range(len(iterable)):
+            if j < len(iterable[i]):
+                result.append(iterable[i][j])
+        j += 1
+
+    return result
+
+# print(interleave('abc', [1, 2, 3], ('!', '@', '#', '$')))
+
+# function to interleave lists
+def interleave_generator(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    while j < longest:
+        for i in range(len(iterable)):
+            if j < len(iterable[i]):
+                yield (iterable[i][j])
+        j += 1
+
+
+# our_generator = interleave_generator('abc', [1, 2, 3], ('!', '@', '#', '$'))
+# for i in our_generator:
+#     print(i)
+
+
+#Harry is rational but not terrible
+"""
+The librarian who was on his way to bring me the coveted copy of "Harry Potter and the Rational Method" stumbled.
+The books flew by and the book chapters scattered everywhere.
+The resources folder contains a compressed file with all the chapters of the story, but unfortunately the name of the files does not match their contents.
+Rename each file so that its new name is a three-digit number that describes the chapter number, followed by the chapter name.
+For example: For the first episode, the file name should be 001 A Day of Very Low Probability.
+During the exercise you may need to combine work with some libraries, including those we have not studied.
+"""
+import os
+import re
+import codecs
+from bs4 import BeautifulSoup
+
+#function get num in string ang make it 3 digits long and return it as a string
+def three_digits(num):
+    if len(num) == 3:
```

Comment:
https://stackoverflow.com/questions/3505831/in-python-how-do-i-convert-a-single-digit-number-into-a-double-digits-string - try to google search it